gobdd.go

```diff
@@ -419,12 +440,28 @@ func (s *Suite) callAfterScenarios(ctx Context) {
 	}
 }
 
+func (s *Suite) callBeforeSteps(ctx Context) {
+	for _, f := range s.options.beforeStep {
+		f(ctx)
+	}
+}
+
+func (s *Suite) callAfterSteps(ctx Context) {
+	for _, f := range s.options.afterStep {
+		f(ctx)
+	}
+}
+
 func (s *Suite) runScenario(ctx Context, scenario *msgs.GherkinDocument_Feature_Scenario,
 	bkg *msgs.GherkinDocument_Feature_Background, t *testing.T) {
-	s.callBeforeScenarios(ctx)
-
-	defer s.callAfterScenarios(ctx)
 	t.Run(fmt.Sprintf("%s %s", strings.TrimSpace(scenario.Keyword), scenario.Name), func(t *testing.T) {
+		// NOTE consider passing t as argument to scenario hooks
+		ctx.Set(TestingTKey{}, t)
+		defer ctx.Set(TestingTKey{}, nil)
+
+		s.callBeforeScenarios(ctx)
+		defer s.callAfterScenarios(ctx)
```

Comment:
❗ This is an important change. Scenario hooks need access to the current scenario testing object and not the feature one.